File: playwright-csharp/tests/Data/DataBuilders/RestfulBooker/AuthPayloadBuilder.cs
using System.Text.Json.Nodes;
using TestTracks.Playwright.CSharp.Specs.Configuration;

namespace TestTracks.Playwright.CSharp.Specs.Data.DataBuilders.RestfulBooker;

public static class AuthPayloadBuilder
{
    public static JsonObject FromCredential(CredentialTarget credential)
    {
        return new JsonObject
        {
            ["username"] = credential.Username,
            ["password"] = credential.Password
        };
    }
}

File: playwright-java/src/test/java/testtracks/data/builders/restfulbooker/AuthPayloadBuilder.java
package testtracks.data.builders.restfulbooker;

import com.fasterxml.jackson.databind.node.JsonNodeFactory;
import com.fasterxml.jackson.databind.node.ObjectNode;
import testtracks.configuration.CredentialTarget;

public final class AuthPayloadBuilder {
  private AuthPayloadBuilder() {}

  public static ObjectNode fromCredential(CredentialTarget credential) {
    var payload = JsonNodeFactory.instance.objectNode();
    payload.put("username", credential.username());
    payload.put("password", credential.password());
    return payload;
  }
}
